bsi/src/main/java/org/roaringbitmap/bsi/longlong/Roaring64NavigableMapSliceIndex.java

```diff
@@ -0,0 +1,710 @@
+    this.minValue = 0;
+    this.ebM = null;
+    this.bA = null;
+  }
+
+  /**
+   * Serialize this bitmapsliceindex(bsi).
+   * <p>
+   * There is no specification for now: it may change from one java version to
+   * another. This serialization format is compatible with bsi of roaring(Go)
+   * @see <a href="https://github.com/RoaringBitmap/roaring/blob/b32ae1a8cc386adbf7753251b8a580ed37f270a9/roaring64/bsi64.go#L882">bsi64.WriteTo</a>
+   * Current serialization format:
+   * ---
+   * ebM: as standard Roaring64NavigableMap format (with specify legacy/portable format)
+   * loop over bA[]:  as standard Roaring64NavigableMap format (with specify legacy/portable format)
+   * ---
+   *
+   * <p>
+   * Consider calling {@link #runOptimize} before serialization to improve compression.
+   * <p>
+   * The current bsi is not modified.
+   *
+   * @param output the DataOutput stream
+   * @throws IOException Signals that an I/O exception has occurred.
+   */
+  public void serialize(DataOutput output) throws IOException {
+    this.ebM.serialize(output);
+    for (Roaring64NavigableMap rb : this.bA) {
+      rb.serialize(output);
+    }
+  }
+
+  public void serialize(ByteBuffer buffer) throws IOException {
+    this.ebM.serializePortable(buffer);
+    for (Roaring64NavigableMap rb : this.bA) {
+      rb.serializePortable(buffer);
+    }
+  }
+
+  /**
+   * Deserialize (retrieve) this bitmapsliceindex(bsi).
+   * <p>
+   * The default serialization format for underlying Roaring64NavigableMap is
+   * {@link Roaring64NavigableMap#SERIALIZATION_MODE_PORTABLE}
+   * <p>
+   * The current bsi is overwritten.
+   *
+   * @param in the DataInput stream
+   * @throws IOException Signals that an I/O exception has occurred.
+   */
+  public void deserialize(DataInput in) throws IOException {
+    this.clear();
+
+    // read ebm
+    Roaring64NavigableMap ebm = new Roaring64NavigableMap();
+    // set serialization format before deserialize
+    ebm.setSerializationMode(serializationMode);
+    ebm.deserialize(in);
+    this.ebM = ebm;
+
+    List<Roaring64NavigableMap> baList = new ArrayList<>();
+    try (DataInputStream is = new DataInputStream((InputStream) in); ) {
+      while (is.available() > 0) {
+        Roaring64NavigableMap rb = new Roaring64NavigableMap();
+        // set serialization format before deserialize
+        rb.setSerializationMode(serializationMode);
+        rb.deserialize(in);
+        baList.add(rb);
+      }
+    }
+    this.bA = baList.toArray(new Roaring64NavigableMap[0]);
+  }
+
+  public void deserialize(ByteBuffer buffer) throws IOException {
+    this.clear();
+    Roaring64NavigableMap ebm = new Roaring64NavigableMap();
+    ebm.setSerializationMode(serializationMode);
+    ebm.deserialize(buffer);
+    this.ebM = ebm;
+    // read back
+    buffer.position(buffer.position() + (int) ebm.serializedSizeInBytes());
+    List<Roaring64NavigableMap> baList = new ArrayList<>();
+    while (buffer.hasRemaining()) {
```

Comment:
I feel we should add a clear note about this deserialization strategy. Am I correct assuming this is fine loading from a file holding a single BSI, but it would fail if one would have written 2 BSI to the same file?

- This could be fixed easily by pre-prending the number of slices, hence not relying on `.hasRemaining`
- This would not be compatible with current Go implementation.
- All these is matter regarding [RoaringBitmapSpec](https://github.com/RoaringBitmap/RoaringFormatSpec) future matter, though I feel important to document current format.

Reply:
this could be fix like what you said, but it would not be compatible with current go impl. 
but when we talk about the [RoaringBitmapSpec](https://github.com/RoaringBitmap/RoaringFormatSpec), i will add some meta info during serialization.

Reply:
Thanks for adding details in current PR.

Reply:
unfortunately, after i re-organized the *NEW* serialization format (adding some meta info, adding number of slices), It will not be compatible with existing BSI implementation (roaring(Go)).

My original intention was to fix the bsi serialization issues between Java and Go implementations. but now it could not.